crates/libcontainer/src/container/container_checkpoint.rs

```diff
@@ -0,0 +1,121 @@
+use super::{Container, ContainerStatus};
+use crate::container::container::CheckpointOptions;
+use anyhow::{bail, Context, Result};
+
+use libcgroups::common::{
+    CgroupSetup::{Hybrid, Legacy},
+    DEFAULT_CGROUP_ROOT,
+};
+use oci_spec::runtime::Spec;
+use std::os::unix::io::AsRawFd;
+
+const CRIU_CHECKPOINT_LOG_FILE: &str = "dump.log";
+
+impl Container {
+    pub fn checkpoint(&mut self, opts: &CheckpointOptions) -> Result<()> {
+        self.refresh_status()
+            .context("failed to refresh container status")?;
+
+        // can_pause() checks if the container is running. That also works for
+        // checkpoitning. is_running() would make more sense here, but let's
+        // just reuse existing functions.
+        if !self.can_pause() {
+            bail!(
+                "{} could not be checkpointed because it was {:?}",
+                self.id(),
+                self.status()
+            );
+        }
+
+        let mut criu = rust_criu::Criu::new().unwrap();
+
+        // We need to tell CRIU that all bind mounts are external. CRIU will fail checkpointing
+        // if it does not know that these bind mounts are coming from the outside of the container.
+        // This information is needed during restore again. The external location of the bind
+        // mounts can change and CRIU will just mount whatever we tell it to mount based on
+        // information found in 'config.json'.
+        let source_spec_path = self.bundle().join("config.json");
+        let spec = Spec::load(&source_spec_path)?;
+        let mounts = spec.mounts().clone();
+        for m in mounts.unwrap() {
+            match m.typ().as_deref() {
+                Some("bind") => {
+                    let dest = m
+                        .destination()
+                        .clone()
+                        .into_os_string()
+                        .into_string()
+                        .unwrap();
```

Comment:
We generally try to avoid using unwrap. 
```suggestion
                        .expect("failed to convert mount destination")
```